cluster/src/main/scala/io/snappydata/impl/LeadImpl.scala

```diff
@@ -183,6 +183,10 @@ class LeadImpl extends ServerImpl with Lead with Logging {
             logInfo("Primary lead lock acquired.")
           // let go.
           case false =>
+            if (!_directApiInvoked) {
+              throw new SQLException("Primary Lead node (Spark Driver) is already running in the system. " +
+                  "You may use split cluster mode to connect to SnappyData cluster.")
+            }
```

Comment:
I think SQLException is inappropriate here and at other places in LeadImpl. Better option is SparkException.